frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -491,7 +511,7 @@ function ReaderBookmark:onShowBookmark()
                             table.remove(item_table, i)
                         end
                     end
-                    bm_menu:switchItemTable(self.filtered_mode and _("Bookmarks (filtered)") or _("Bookmarks"), item_table, -1)
+                    bm_menu:switchItemTable(bookmark.filtered_mode and _("Bookmarks (search results)") or _("Bookmarks"), item_table, -1)
```

Comment:
It just didn't work with `self` after exiting from the select mode.

Reply:
OK, I guess that in these methods that are dynamically added, there is a `self` added implicitely by Lua because of the `:` that would then mask/shadow the upper self. That's why we give it a second name `bookmark` so this one is not shadowed by the implicite `self`.
```lua
function ReaderBookmark:onShowBookmark()
    self.select_mode = false -- this "self" is the ReaderBookmark instance
    [...]
    local bookmark = self
    [...]
    function bm_menu:onMenuHold(item)
        if self.select_mode then -- this "self" is the Menu instance
```